src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2200,6 +2208,9 @@ def aks_agentpool_update(cmd,   # pylint: disable=unused-argument
         instance.min_count = int(min_count)
         instance.max_count = int(max_count)
 
+    if max_surge:
+        instance.upgrade_settings = AgentPoolUpgradeSettings(max_surge=max_surge)
```

Comment:
this will replace the whole setting. Maybe only update the max surge value so the old version cli extension won't break any future change like max unavailable? instance.upgrade_settings.max_surge = max_surge

Reply:
ensured upgrade_settigns wasn't null and just set max_surge